src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -330,11 +343,16 @@ private NotificationMessage doSync(FormplayerSyncScreen screen) throws Exception
      */
     private NotificationMessage doQuery(FormplayerQueryScreen screen,
                                         MenuSession menuSession,
-                                        Hashtable<String, String> queryDictionary) throws CommCareSessionException {
+                                        Hashtable<String, String> queryDictionary,
+                                        boolean autoSearch) throws CommCareSessionException {
         log.info("Formplayer doing query with dictionary " + queryDictionary);
         NotificationMessage notificationMessage = null;
-        screen.answerPrompts(queryDictionary);
-        String responseString = queryRequester.makeQueryRequest(screen.getUriString(), restoreFactory.getUserHeaders());
+
+        if (queryDictionary != null) {
```

Comment:
I think the null checking may no longer be necessary now that web apps is fixed to not send a null value. But I suppose it's better defensive coding to keep the checks.